File: mobile/src/models/SettlementMember.ts
export interface SettlementMember {
  id: string;
  settlementId: string;
  userId: string;
  role: 'OWNER' | 'MEMBER';
  joinedAt: string;
}

export interface InviteCodeResponse {
  code: string;
  expiresAt: string;
  settlementId: string;
}

export interface JoinByCodeRequest {
  code: string;
}

File: backend/src/main/java/com/settleup/domain/settlement/SettlementMember.java
package com.settleup.domain.settlement;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "settlement_members",
        uniqueConstraints = @UniqueConstraint(columnNames = {"settlement_id", "user_id"}))
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SettlementMember {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @Column(name = "settlement_id", nullable = false)
    private UUID settlementId;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private MemberRole role = MemberRole.MEMBER;

    @Column(name = "joined_at", nullable = false)
    private LocalDateTime joinedAt;

    @PrePersist
    protected void onCreate() {
        if (joinedAt == null) {
            joinedAt = LocalDateTime.now();
        }
    }

    public enum MemberRole {
        OWNER, MEMBER
    }
}
